### ISSUE TITLE
Neo4j Updates

### ISSUE THREAD

**Issue by 0x6d6f7468** [PlumHound/PlumHound#43]
Need to update Neo4j to a newer version.

Issue: Python3.9 installs fail on modern Kali fails due to distutils errors. Installation of Neo4j version 1.7 (frozen version in requirements.txt) fails on Python3.10 because the Mapping module was moved from collections to collections.abc. We are now between a rock and a hard place regarding Neo4j, and the need to upgrade has come up once again.

Issue is related to #36. Given BloodHound recommends the usage of Neo4j 4.4.13, we should look into using version 4.4. Not sure how systemic updates will need to be, but will look into it as I have time.

**Comment by 0x6d6f7468** [comment]
Offending code appears to be https://github.com/PlumHound/PlumHound/blob/master/lib/phTasks.py#L179 (or at least that's where the problems started).

Quick aside: We should consider adding more robust error messages within exceptions. Currently, vanilla output logs with max verbosity don't give any relevant information.

Anyway, looks like `results.detach()` is no longer a method in Neo4j. While researching the error, came across this link: https://neo4j.com/docs/upgrade-migration-guide/current/version-4/migration/drivers/python-driver/

I did a quick grep over the code and it looks like detach is the only problematic function. If we can replace its usage with `Results.consume()`, we should be golden. I'll keep looking into it.

(Also, that article also indicates that we can drop neobolt and neotime from requirements.txt)

**Comment by Relkci** [comment]
nice work by the community updating this to current gen python and neo4j drivers.  #44  thanks much!
